packages/smart-contracts/src/contracts/BatchPayments.sol

```diff
@@ -12,56 +13,54 @@ import '@openzeppelin/contracts/security/ReentrancyGuard.sol';
  * @notice This contract makes multiple payments with references, in one transaction:
  *          - on: ERC20 Payment Proxy and ETH Payment Proxy of the Request Network protocol
  *          - to: multiple addresses
- *          - fees: Proxy fees can be paid to the same address.
- *                  An additional batch fee is added, paid to the same address.
+ *          - fees: ERC20 and ETH proxies fees are paid to the same address.
+ *                  An additional batch fee is paid to the same address.
+ *         If one transaction of the batch fail, every transactions are reverted.
  */
 contract BatchPayments is Ownable, ReentrancyGuard {
-  // This event is declared in ERC20FeeProxy
-  event TransferWithReferenceAndFee(
-    address tokenAddress,
-    address to,
-    uint256 amount,
-    bytes indexed paymentReference,
-    uint256 feeAmount,
-    address feeAddress
-  );
+  using SafeERC20 for IERC20;
+
+  IERC20FeeProxy public paymentErc20FeeProxy;
+  IEthereumFeeProxy public paymentEthFeeProxy;
 
   // @dev: Between 0 and 1000, i.e: batchFee = 10 represent 1% of fee
   uint256 public batchFee;
 
-  IERC20FeeProxy public paymentErc20FeeProxy;
-  IEthereumFeeProxy public paymentEthereumFeeProxy;
+  struct Token {
+    address tokenAddress;
+    uint256 amountAndFee;
+    uint256 batchFeeAmount;
+  }
 
   /**
    * @param _paymentErc20FeeProxy The address to the ERC20 payment proxy to use.
-   * @param _paymentEthereumFeeProxy The address to the Ethereum payment proxy to use.
+   * @param _paymentEthFeeProxy The address to the Ethereum payment proxy to use.
    * @param _owner Owner of the contract.
    */
   constructor(
     address _paymentErc20FeeProxy,
-    address _paymentEthereumFeeProxy,
+    address _paymentEthFeeProxy,
     address _owner
   ) {
     paymentErc20FeeProxy = IERC20FeeProxy(_paymentErc20FeeProxy);
-    paymentEthereumFeeProxy = IEthereumFeeProxy(_paymentEthereumFeeProxy);
-    batchFee = 0;
+    paymentEthFeeProxy = IEthereumFeeProxy(_paymentEthFeeProxy);
     transferOwnership(_owner);
+    batchFee = 0;
   }
 
-  // Needed because batchEthPaymentsWithReferenceAndFee requires that the contract has ethers
+  // batch Eth requires batch contract to receive the funds in order to send it to recipients
   receive() external payable {}
```

Comment:
Should we protect the user that sends ETH directly to the contract address without using the contract methods?
Like so:
https://github.com/RequestNetwork/requestNetwork/blob/53e587916e336057935c3d54655de0dc539a59a3/packages/smart-contracts/src/contracts/EthereumFeeProxy.sol#L20-L23

Reply:
I cannot do that because the contract needs to receive funds, it is called in batch ETH function.
(I revert "receive", it fails)